### PR TITLE
Get privatedns.sh refactor

### PR DESCRIPTION
**Pull request by vfrazao-ns1**
This is a re-write of `get_privatedns.sh` which has a couple of issues around downloading the same image multiple times and attempting to download images that don't exist (e.g., web container v2.2.2).

This is written in Python and has been tested to work on Python 2 and 3 using only standard library modules. The script has been tested on Mac and Linux, it mot likely does not work on Windows.

### CODE REVIEW

**Comment by vfrazao-ns1** [comment]
One other thing to figure out is what to do with the existing `get_privatedns.sh` script. We can:

1. Place it in an archive folder
2. Remove it entirely
3. Modify it to download and invoke the Python script

Doing 1. or 2. will also require modifying our installation guide to reference the new script.

**Comment by jwilner** [comment]
> One other thing to figure out is what to do with the existing `get_privatedns.sh` script. We can:
> 
> 1. Place it in an archive folder
> 2. Remove it entirely
> 3. Modify it to download and invoke the Python script
> 
> Doing 1. or 2. will also require modifying our installation guide to reference the new script.

Thanks Victor. I'm going to add @ponbiki to this PR too since he worked some on the old form.

**Comment by jwilner** [comment]
NM I can't add -- can you add Jandy?

**Comment by vfrazao-ns1** [comment]
> NM I can't add -- can you add Jandy?

Added Jandy to the review

File: utils/get_privatedns/get_privatedns.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+        "headers": headers,
+    }
+
+
+def make_progress_bar(completed, total, complete=False):
+    # type: (int, int, Optional[bool]) -> None
+    """
+    Prints a progress bar to stdout
+    """
+    bar_size = 80
```
Comment (ponbiki) [inline utils/get_privatedns/get_privatedns.py]: You are one motivated coder, and a half-step away from a curses app)

**Pull request merged**